param/parameterized.py

```diff
@@ -519,26 +519,38 @@ def __setstate__(self,state):
 # Define one particular type of Parameter that is used in this file
 class String(Parameter):
     """
-    A simple String parameter.
+    A String Parameter, with a default value and optional regular expression.
+
+    Example of creating a String:
+      ip_regexp = '^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$'
+      IP = String(default='0.0.0.0', regexp=ip_regexp, doc='An IP address.')
```

Comment:
Good example, though in practice I'd probably recommend making a subclass in this case if you are using IP addresses in many locations.  E.g.:

```
class IPAddress(String):
    "IPV4 address as a string"
    def __init__(self, default="0.0.0.0", **kwargs):
       super(IPAddress, self).__init__(default=default, regexp=ip_regexp, **kwargs)
```

That way you declare that your String is actually an IP address, which won't be evident otherwise. Plus it lets you define what the default value is.